khmer/_khmermodule.cc

```diff
@@ -3758,42 +3754,35 @@ static PyObject * khmer_labelhash_new(PyTypeObject *type, PyObject *args,
     self = (khmer_KLabelHash_Object*)type->tp_alloc(type, 0);
 
     if (self != NULL) {
-        WordLength k = 0;
-        PyListObject * sizes_list_o = NULL;
+        PyObject * hashtable_o;
+        khmer::Hashtable * hashtable = NULL;
 
-        if (!PyArg_ParseTuple(args, "bO!", &k, &PyList_Type, &sizes_list_o)) {
+        if (!PyArg_ParseTuple(args, "O", &hashtable_o)) {
             Py_DECREF(self);
             return NULL;
         }
 
-        std::vector<HashIntoType> sizes;
-        Py_ssize_t sizes_list_o_length = PyList_GET_SIZE(sizes_list_o);
-        for (Py_ssize_t i = 0; i < sizes_list_o_length; i++) {
-            PyObject * size_o = PyList_GET_ITEM(sizes_list_o, i);
-            if (PyLong_Check(size_o)) {
-                sizes.push_back((HashIntoType) PyLong_AsUnsignedLongLong(size_o));
-            } else if (PyInt_Check(size_o)) {
-                sizes.push_back((HashIntoType) PyInt_AsLong(size_o));
-            } else if (PyFloat_Check(size_o)) {
-                sizes.push_back((HashIntoType) PyFloat_AS_DOUBLE(size_o));
-            } else {
-                Py_DECREF(self);
-                PyErr_SetString(PyExc_TypeError,
-                                "2nd argument must be a list of ints, longs, or floats");
-                return NULL;
-            }
+        if (PyObject_TypeCheck(hashtable_o, &khmer_KHashbits_Type)) {
+          khmer_KHashbits_Object * kho = (khmer_KHashbits_Object *) hashtable_o;
+          hashtable = kho->hashbits;
+        }
+        else if (PyObject_TypeCheck(hashtable_o, &khmer_KCountingHash_Type)) {
+          khmer_KCountingHash_Object * cho = (khmer_KCountingHash_Object *) hashtable_o;
+          hashtable = cho->counting;
+        }
+        else {
+          PyErr_SetString(PyExc_ValueError,
+                          "graph object must be a NodeGraph or CountGraph");
+          Py_DECREF(self);
+          return NULL;
```

Comment:
How odd. I verified that I got exactly that error message, and that it's present nowhere else in the code base, so it's covered!